sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TimeWindow.scala

```diff
@@ -35,6 +35,10 @@ case class TimeWindow(
   with ImplicitCastInputTypes
   with Unevaluable
   with NonSQLExpression {
+  require(windowDuration > 0, "The window duration must be " +
+    s"a positive integer, long or string literal, found: $windowDuration")
+  require(slideDuration > 0, "The slide duration must be " +
+    s"a positive integer, long or string literal, found: $slideDuration")
```

Comment:
We cannot check these requirements only in the constructor?

Reply:
The constructor has no information about the unit of the duration e.g. -1 day would get converted to -86,400,000,000 in the error message, which is confusing.

Reply:
Either way, we'd be better to avoid duplicate error checks. Can we make it simpler, e.g., how about making a helper func to check these requirements?

Reply:
In scala, this() call has to be first statement in a constructor, this limits the extent to which code deduplication can be done. I have made changes to use a helper function for other instances.